documentation/src/main/asciidoc/reference/troubleshooting.asciidoc

```diff
@@ -0,0 +1,103 @@
+[[troubleshooting]]
+= Troubleshooting
+
+[[troubleshooting-under-the-hood]]
+== Finding out what is executed under the hood
+
+For search queries, you can get a human-readable representation of a <<search-dsl-query-object,`SearchQuery` object>>
+by calling `toString()` or `queryString()`.
+Alternatively, rely on the logs:
+<<troubleshooting-logging-query,`org.hibernate.search.query`>> will log every query at the `DEBUG` level
+before it is executed.
+
+For more general information about what is being executed, rely on loggers:
+
+* <<troubleshooting-logging-lucene-infostream,`org.hibernate.search.backend.lucene.infostream`>>
+for Lucene.
+* <<troubleshooting-logging-elasticsearch-request,`org.hibernate.search.elasticsearch.request`>>
+for Elasticsearch.
+
+[[troubleshooting-logging]]
+== Loggers
+
+Here are a few loggers that can be useful when debugging an application that uses Hibernate Search:
+
+[[troubleshooting-logging-query]]`org.hibernate.search.query`::
+Available for all backends.
++
+Logs every single search query at the `DEBUG` level, before its execution.
+[[troubleshooting-logging-elasticsearch-request]]`org.hibernate.search.elasticsearch.request`::
+Available for Elasticsearch backends only.
++
+Logs every single request sent to Elasticsearch at the `DEBUG` or `TRACE` level after its execution.
+Use the `DEBUG` level for compact information (method, path, execution time, status code, ...),
+or the `TRACE` level to also get the full request and response.
++
+You can enable pretty-printing (line breaks and indentation) for the request and response
+using a <<backend-elasticsearch-configuration-logging,configuration property>>.
+[[troubleshooting-logging-lucene-infostream]]`org.hibernate.search.backend.lucene.infostream`::
+Available for Lucene backends only.
++
+Logs low level trace information about Lucene's internal components, at the `TRACE` level.
++
+Enabling the `TRACE` level on this logger is not enough:
+you must also enable the infostream using a <<backend-lucene-io-writer-infostream,configuration property>>.
```

Comment:
/using a/using the/s ?

Reply:
No, using a configuration property. There are many.